File: db4o.net/Db4oUnit.Extensions/Db4oUnit.Extensions/Tests/UnhandledExceptionInThreadTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4oUnit;
using Db4oUnit.Extensions;
using Db4oUnit.Extensions.Fixtures;
using Db4oUnit.Extensions.Tests;
using Db4objects.Db4o.Internal;
using Sharpen.Lang;

namespace Db4oUnit.Extensions.Tests
{
	public class UnhandledExceptionInThreadTestCase : ITestCase
	{
		public class ExceptionThrowingTestCase : AbstractDb4oTestCase
		{
			public virtual void Test()
			{
				Container().ThreadPool().Start(ReflectPlatform.SimpleName(typeof(UnhandledExceptionInThreadTestCase
					)) + " Throwing Exception Thread", new _IRunnable_15());
			}

			private sealed class _IRunnable_15 : IRunnable
			{
				public _IRunnable_15()
				{
				}

				public void Run()
				{
					throw new InvalidOperationException();
				}
			}
		}

		public virtual void TestSolo()
		{
			Db4oTestSuiteBuilder suite = new Db4oTestSuiteBuilder(new Db4oInMemory(), typeof(
				UnhandledExceptionInThreadTestCase.ExceptionThrowingTestCase));
			TestResult result = new TestResult();
			new TestRunner(suite).Run(result);
			Assert.AreEqual(1, result.Failures.Count);
		}
	}
}

File: db4ounit.extensions/src/db4ounit/extensions/tests/UnhandledExceptionInThreadTestCase.java
/* Copyright (C) 2009  Versant Inc.  http://www.db4o.com */

package db4ounit.extensions.tests;

import com.db4o.internal.*;

import db4ounit.*;
import db4ounit.extensions.*;
import db4ounit.extensions.fixtures.*;

public class UnhandledExceptionInThreadTestCase implements TestCase {
	
	public static class ExceptionThrowingTestCase extends AbstractDb4oTestCase {
		public void test() {
			container().threadPool().start(ReflectPlatform.simpleName(UnhandledExceptionInThreadTestCase.class)+" Throwing Exception Thread", new Runnable() {
				public void run() {
					throw new IllegalStateException();
				}
			});
		}
	}
	
	public void testSolo() {
		
		final Db4oTestSuiteBuilder suite = new Db4oTestSuiteBuilder(new Db4oInMemory(), ExceptionThrowingTestCase.class);
		final TestResult result = new TestResult();
		new TestRunner(suite).run(result);
		Assert.areEqual(1, result.failures().size());
		
	}
}
